.github/workflows/doc_build.yml

```diff
@@ -0,0 +1,28 @@
+# Build Documentation when new release created
```

Comment:
To make it more clear, let's name this workflow and the file name, `doc_deploy` as it is only used when there is a release.